### PR TITLE
Fix/#800 - Run core service twice should raise an error

### PR DESCRIPTION
**Pull request by trgiangdo**
https://github.com/Avaiga/taipy-core/issues/800

### CODE REVIEW

**Review by toan-quach — DISMISSED** [review]
lgtm :D

File: src/taipy/core/_core.py [hunk 1]
```diff
@@ -21,17 +22,21 @@
...
     Core service
     """
 
+    __is_running = False
+
     __logger = _TaipyLogger._get_logger()
 
     _orchestrator: Optional[_Orchestrator] = None
     _dispatcher: Optional[_JobDispatcher] = None
+    _lock = Lock()
```
Comment (jrobinAV) [inline src/taipy/core/_core.py]: Could you move the lock close to the `__is_running` attribute? Maybe we can rename it `_lock_is_running` as well.

File: tests/core/notification/test_events_published.py [hunk 2]
```diff
@@ -40,6 +41,11 @@ def identity(x):
 
 
 def test_event_published():
+    if _OrchestratorFactory._orchestrator is None:
+        _OrchestratorFactory._build_orchestrator()
+    _OrchestratorFactory._build_dispatcher()
+    Config.unblock_update()
+
```
Comment (jrobinAV) [inline tests/core/notification/test_events_published.py]: I wonder why this is needed.

File: tests/core/test_core.py [hunk 3]
```diff
@@ -70,6 +74,27 @@ def test_run_core_as_a_service_standalone_mode(self):
         assert isinstance(_OrchestratorFactory._dispatcher, _StandaloneJobDispatcher)
         assert core._dispatcher.is_running()
         assert _OrchestratorFactory._dispatcher.is_running()
+        core.stop()
+
+    def test_core_service_can_only_be_run_once(self):
+        core_instance_1 = Core()
+        core_instance_2 = Core()
+        Config.configure_job_executions(mode=JobConfig._DEVELOPMENT_MODE)
```
Comment (jrobinAV) [inline tests/core/test_core.py]: This line should not be necessary. The conftest should have cleaned the Config. Am I right?

**Review by jrobinAV — DISMISSED** [review]
Looks good to me

File: tests/core/test_core.py (hunk 3 above)
Comment (trgiangdo) [reply tests/core/test_core.py]: Yes, we can remove those

File: tests/core/notification/test_events_published.py (hunk 2 above)
Comment (trgiangdo) [reply tests/core/notification/test_events_published.py]: My bad. I believe this is redundant from the debugging process. I will remove it and try to rerun the tests.

File: tests/core/notification/test_events_published.py [hunk 4]
```diff
@@ -12,6 +12,7 @@
 from queue import SimpleQueue
 
 from src.taipy.core import taipy as tp
+from src.taipy.core._orchestrator._orchestrator_factory import _OrchestratorFactory
```
Comment (jrobinAV) [inline tests/core/notification/test_events_published.py]: I don't think we need it.

**Review by jrobinAV — DISMISSED** [review]

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**